File: app/core/src/main/java/com/teamg/tourdeshot/core/model/Local.java
package com.teamg.tourdeshot.core.model;

import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.Transient;
import org.springframework.data.mongodb.core.index.GeoSpatialIndexed;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@Data
@NoArgsConstructor
@Document(collection = "locals")
public class Local {

    @Transient
    public static final String SEQUENCE_NAME = "locals_sequence";

    @Id
    private Long id;

    private String name;

    private Long ownerId;

    @GeoSpatialIndexed
    private Double[] coordinates;

    private Address address;

    private String image;

    private List<String> localCategories;

    private Integer priceCategory;

    private OpeningHours openingHours;

    private String website;

    private Contact contact;

    private Menu menu;

}

File: ui/src/app/models/Local.ts
import { Address } from './Address';
import { Details } from './Details';
import { Menu } from './Menu';
import { Coordinates } from './Coordinates';

export interface Local {
  id: number;
  name: string;
  ownerId: number;
  coordinates: Coordinates;
  address: Address;
  details: Details;
  menu: Menu;
  distance: number;
}
